checkstyle/src/main/java/org/infinispan/checkstyle/checks/ForbiddenMethodCheck.java

```diff
@@ -0,0 +1,85 @@
+package org.infinispan.checkstyle.checks;
+
+import static com.puppycrawl.tools.checkstyle.api.TokenTypes.METHOD_CALL;
+
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Check that will log an error when a method that matches the given regular expression is invoked. This method
+ * can optionally on match a method invocation based on the number of arguments provided to it. This check does not
```

Comment:
Typo: on match

I don't think individual checks should care about production vs test code, it's the job of the configuration to exclude test sources.

I'd say
```
Checks that certain methods are not invoked

The methodPattern argument is a regular expression, invocations of method names matching the pattern are forbidden.
Only method names are checked, the defining class doesn't matter.

If the optional argumentCount argument is set, only overloads with that number of parameters are forbidden.
Forbidding only overloads with 0 arguments is not supported at this time.

@author
@since
```

Reply:
Sure, sounds good.